src/main/java/webserver/JsonTodos.java

```diff
@@ -14,29 +12,29 @@
 @SuppressWarnings("unchecked")
 public class JsonTodos implements Todos {
```

Comment:
I didn't catch this earlier, but this looks like a good application of introducing an interface (or base class).  Now you can get TODO items from JSON files or just about anywhere else.

Take a moment and notice a few properties about the interface and its subclasses

* Adding additional implementations doesn't require the interface to change.  For this to happen, the methods, parameters, and return values in the interface methods have to be abstract enough to survive the addition of the next implementation class.  You're returning things like `JSONObject` (a class you wrote) and take in things like `int` IDs.  Those are pretty good abstractions -- give me an ID; I'll give you a TODO item.
* Different implementations do need different things.  You can't change the interface, so this information has to be provided in the constructors of the implementations.  Hence the `todosFile` constructor argument, which I suspect is not there in other implementations.
* TL;DR of all that: a stable interface deals strictly in the domain (using primitives or defining its own types as necessary), while implementations often differ in their constructor arguments.

The only thing I'd consider changing later would be to have a domain-centric notion of a TODO item (you use `JSONObject` and `JSONArray` so far in the types returned by these methods, which implies reliance on JSON).  So later on you might have a model class called `TODOItem` that may (or may not) be backed by a JSON data structure.  `JSONTODOItem` might then extend that class (or implement that interface).  Then this interface could deal in IDs and TODO items and stand the test of time.